README.md

```diff
@@ -51,87 +48,328 @@ console.log(`Running on ${await server.httpUrl()}`);
+}
+
+type Subscription {
+  newItem: Item!
+  updatedItem: Item!
+  deletedItem: ID!
+}
+```
 
-  ```js
-  // see the example ./examples/apollo-client/test/mutations.spec.ts for the full code
+#### `getDatabaseSchema(): Promise<{[table: string]: string[]}>`
 
-  await client.mutate({
-    mutation: CREATE_ITEM,
-    variables: { title: "TestA" }
-  });
+Return the generated database schema.
 
-  const result = await client.query({
-    query: FIND_ALL_ITEMS,
-    fetchPolicy: "network-only"
-  });
+```js
+// server: TestxServer
+await server.bootstrap();
+console.log(await server.getDatabaseSchema());
+```
 
-  expect(result.data.findAllItems[0].title).to.be.equal("TestA");
-  ```
+```js
+{
+  item: ["id", "name", "created_at", "updated_at"];
+}
+```
 
-- Auto generated client queries, mutations and subscriptions
+#### `getQueries(): Promise<{[query: string]: string}>`
 
-- Rich API to control the server execution and database state
-  [ `start()`, `stop()`, `close()`, `cleanDatabase()`, ... ]
+Return ready-to-use client queries.
 
-## Examples
+```js
+// server: TestxServer
+await server.bootstrap();
+console.log((await server.getQueries()).findAllItems);
+```
 
-- Testing **Apollo Client** library with **mocha**
+```graphql
+query findAllItems {
+  findAllItems {
+    ...ItemFields
+  }
+}
+
+fragment ItemFields on Item {
+  id
+  name
+}
+```
 
-  [graphql-testx-apollo-client-example](./examples/apollo-client)
+#### `getMutations(): Promise<{[query: string]: string}>`
 
-- Testing **Offix** library with **Jest**
+Return ready-to-use client mutations like for `getQueries()`.
 
-  [graphql-testx-offix-example](./examples/offix)
+#### `getSubscriptions(): Promise<{[query: string]: string}>`
 
-- Testing generic application with **Karma**
+Return ready-to-use client subscriptions like for `getQueries()`.
 
-  [graphql-testx-karma-example](./example/karma)
+### `TestxController`
+
+Controls the `TestxServer` remotely using the `TestxDirector`.
+
+The `TestxController` is used in combination with the `TestxDirector` in the
+[Karma example](./example/karma) in order to start/stop the server, and clean
+the database from the browser.
+
+```
+ TestxDirector.start()     | Browser
+          |
+POST / { name: "start" }
+          |
+    TestxController        |
+          |                | Node.js
+  TestxServer.start()      |
+```
+
+```js
+import { TestxServer, TestxController } from "graphql-testx";
+
+// server: TestxServer
+const controller = new TestxController(server);
+await controller.start();
+
+console.log(`Controller running on ${await controller.httUrl()}`);
+
+await controller.close();
+```
+
+#### `start(port?: number): Promise<void>`
+
+Start the `TestxController` API to a random available port on localhost or to
+the passed port.
+
+#### `httpUrl(): Promise<string>`
+
+Return the http url to the `TestxController` API.
+
+#### `stop(): Promise<void>`
+
+Stop and close the `TestxController` and the `TestxServer`
+
+### `TestxDirector`
```

Comment:
I'm not getting this. Maybe explaining it better.

Reply:
What do you mean? that's just the name of the class, how could I explain it better?